res/controllers/common-controller-scripts.js

```diff
@@ -28,6 +28,52 @@ String.prototype.toInt = function() {
     return a;
 };
 
+/**
+ * Discriminates whether an object was created using the `{}` synthax.
+ *
+ * Returns true when was an object was created using the `{}` synthax.
+ * False if the object is an instance of a class like Date or Proxy or an Array.
+ *
+ * Object.isSimpleObject({}) // true
+ * Object.isSimpleObject(null) // false
+ * Object.isSimpleObject(undefined) // false
+ * Object.isSimpleObject(new Date) // false
+ * Object.isSimpleObject(new (class {})()) // false
+ * @param {any} obj Object to test
+ * @returns {boolean} true if obj was created using the `{}` or `new Object()` synthax, false otherwise
+ */
+Object.isSimpleObject = function(obj) {
+    return obj !== null && typeof obj === "object" && obj.constructor.name === "Object";
+};
+
+/**
+ * Deeply merges 2 objects (Arrays and Objects only, not Map for instance).
+ * @param target {object | Array} Object to merge source into
+ * @param source {object | Array} Object to merge into source
+ */
+Object.deepMerge = function(target, source) {
+    if (target === source || target === undefined) {
+        return;
+    }
+
+    if (Array.isArray(target) && Array.isArray(source)) {
+        target.push(...source);
```

Comment:
this'll only concatenate the array, right? not merge them. I think you'd have to zip em together and then call deepMerge on each pair of elements.

Reply:
This should be fixed with the last commit. Are there tests for JS anywhere?

Reply:
not sure. some basics ones somewhere burrowed in the controller engine tests somewhere, but idk. I'd be nice to have [Qt Quick test](https://doc.qt.io/qt-6/qtquicktest-index.html) integration into our current test suite to do proper testing.

Reply:
> This should be fixed with the last commit. Are there tests for JS anywhere?

You can implement Mixxx unit test for JavaScript, like I did in https://github.com/mixxxdj/mixxx/pull/12437